ablator/utils/base.py

```diff
@@ -70,16 +144,66 @@ def get_lr(optimizer):
 
 
 def debugger_is_active() -> bool:
-    """Return if the debugger is currently active"""
+    """
+    Check if the debugger is currently active.
+
+    Returns
+    -------
+    bool
+        True if the debugger is active, False otherwise.
+    
+    Notes
+    -----
+    Return if the debugger is currently active
+    """
     gettrace = getattr(sys, "gettrace", lambda: None)
     return gettrace() is not None
 
 
 def get_latest_chkpts(checkpoint_dir: Path) -> list[Path]:
+    """
+    Get a list of the latest checkpoint files in a directory.
+
+    Parameters
+    ----------
+    checkpoint_dir : Path
+        The directory containing checkpoint files.
+
+    Returns
+    -------
+    list[Path]
+        A list of the latest checkpoint files sorted by filename.
+    """
     return sorted(list(checkpoint_dir.glob("*.pt")))[::-1]
 
 
 def parse_device(device: ty.Union[str, list[str]]):
+    """
+    Parse a device string or a list of device strings.
```

Comment:
- We should mention when `ValueError` is raised (when `device` a string but is not either `cuda` or `cpu` cor starts with `cuda:`)
- There's an `if` condition for the case where `device` is an `int`, but it's not included in the param type definition. Maybe raise an Issue about this?
- If `device` is not of any type defined (str, list[str], int) then this class will return `cuda` if available, otherwise `cpu`

Reply:
Already submit a issue.